Automattic/jetpack
# Premium content block: Add a warning message for invalid plans

When a Premium content block is configured with an invalid plan (subscription), the user is not aware that there's a problem with it. To handle this, we can add a warning message that would inform the user that the block is currently in an invalid state.





Fixes https://github.com/Automattic/wp-calypso/issues/49103
#### Changes proposed in this Pull Request:

* Added a warning message at the top of the block
* Changed the plan (subscription) description with the message "Subscription not found".

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* On Atomic, use Jetpack beta to switch to this branch
* On Simple, apply D73590-code on your sandbox
* Create a new Premium Content Block
* Connect it to Stripe (I think you'll need to create a Stripe test account :( )
* Configure the subscription in the controls and save the post
* Use incognito and go to the post you just published. Make sure that when you click on the "Subscribe" button the Stripe payment form is loaded
* Back in the editor, enable the Code Editor
* In the content, replace the "selectedPlanId" attribute with a non-existing plan.
* Exit the Code Editor and you should be able to see a warning message saying that the plan is invalid.

**Before**
![Screenshot 2022-01-24 at 11 23 28](https://user-images.githubusercontent.com/995955/150755917-0c9ee4fc-43ed-4b82-96fc-c93dcff6b25b.png)


**After**
![Screenshot 2022-01-24 at 11 20 41](https://user-images.githubusercontent.com/995955/150755971-1669faa4-6c2c-410a-8074-9cd86ba1c845.png)
